sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_utils.py

```diff
@@ -320,3 +339,66 @@ def _get_url_for_http_request(attributes: Attributes) -> Optional[str]:
                     http_target,
                 )
     return url
+
+def _get_DJB2_sample_score(trace_id_hex: str, sampler_type) -> float:
+    # This algorithm uses 32bit integers
+    hash_value = Int32(_SAMPLING_HASH)
+    for char in trace_id_hex:
+        hash_value = ((hash_value << 5) + hash_value) + ord(char)
+
+    if hash_value == _INTEGER_MIN:
+        hash_value = int(_INTEGER_MAX)
+    else:
+        hash_value = abs(hash_value)
+
+    if sampler_type == "rate_limited":
```

Comment:
What are the different scenarios you want to support here with sampler_type?

Reply:
what is the difference between the results? can you refactor the code here to handle both to simplify this?